src/internal/proc.js

```diff
@@ -12,15 +12,17 @@ export const TASK_CANCEL = {toString() { return '@@redux-saga/TASK_CANCEL' }}
 const matchers = {
   wildcard  : () => kTrue,
   default   : pattern => input => input.type === pattern,
-  array     : patterns => input => patterns.some(p => p === input.type),
+  stringable: pattern => input => input.type === pattern.toString(),
```

Comment:
this one could be completely avoidable with combo of
`default   : pattern => input => input.type === String(pattern),`
+
`: is.stringableFunc(pattern) ? matchers.default`